### PR TITLE
Introduce getEventNameFor and use it in WalletService

### PR DESCRIPTION
**Pull request by JustynaBroniszewska**

### CODE REVIEW

File: universal-login-sdk/src/core/services/WalletService.ts [hunk 1]
```diff
@@ -148,9 +148,11 @@ export class WalletService {
       contractAddress: connectingWallet.contractAddress,
       key: connectingWallet.publicKey,
     };
-    return new Promise((resolve, reject) => {
+    // eslint-disable-next-line no-async-promise-executor
+    return new Promise(async (resolve, reject) => {
       const setWallet = this.setWallet.bind(this);
-      const unsubscribe = this.sdk.subscribe('KeyAdded', filter, () => {
+      const addKeyEvent = await this.sdk.walletContractService.getEventNameFor(connectingWallet.contractAddress, 'KeyAdded');
```
Comment (xitronix) [inline universal-login-sdk/src/core/services/WalletService.ts]: Maybe we shoud move line 154 to 151 and than remove async from promise executor?

Comment (JustynaBroniszewska) [reply universal-login-sdk/src/core/services/WalletService.ts]: ✅

File: universal-login-sdk/src/core/services/WalletService.ts [hunk 2]
```diff
@@ -159,8 +160,9 @@ export class WalletService {
     });
   }
 
-  async cancelWaitForConnection() {
+  async cancelWaitForConnection(tick = 500, timeout = 1500) {
     if (this.state.kind === 'Deployed') return;
+    await waitUntil(() => !!this.getConnectingWallet().unsubscribe, tick, timeout);
```
Comment (xitronix) [inline universal-login-sdk/src/core/services/WalletService.ts]: Can we now remove this line?

Comment (JustynaBroniszewska) [reply universal-login-sdk/src/core/services/WalletService.ts]: still, we can not do it

File: universal-login-sdk/test/core/services/WalletService.int.test.ts [hunk 3]
```diff
@@ -83,7 +83,7 @@ describe('INT: WalletService', async () => {
       await walletService.initializeConnection(ensName);
       walletService.waitForConnection();
```
Comment (xitronix) [inline universal-login-sdk/test/core/services/WalletService.int.test.ts]: And then try to remove from WalletService
```ts
await waitUntil(() => !!this.getConnectingWallet().unsubscribe, tick, timeout);
```

```suggestion
      await walletService.waitForConnection();
```

**Pull request merged**